test/tools/h3_request/h3_request.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+import asyncio
+import sys
+from typing import AsyncIterator, cast
+from urllib.parse import urlparse
+
+import aioquic
+from aioquic.asyncio.protocol import QuicConnectionProtocol
+from aioquic.h3.connection import H3_ALPN, H3Connection
+from aioquic.h3.events import (
+    DataReceived,
+    H3Event,
+    HeadersReceived,
+)
+from aioquic.quic.events import QuicEvent
+
+CA_PATH = "test/config/integration/certs/cacert.pem"
+
+
+class Http3Client(QuicConnectionProtocol):
+    """Note, this class is extremely minimal.
+
+    It supports only GET, doesn't properly validate URLs, etc. Since this
+    is just for tests, that's all that's required right now.
+    It is based on https://github.com/aiortc/aioquic/blob/main/examples/http3_client.py
+    which is a far more complete implementation.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self._http = H3Connection(self._quic)
```

Comment:
i tend to use `@cached_property` for these types of props - but will let you decide on that